### ISSUE TITLE
LAPACKE_cheev_work(LAPACK_ROW_MAJOR,'V',...) does not correctly transpose output eigenvectors 

### ISSUE THREAD

**Issue by francis36** [Reference-LAPACK/lapack#379]
Following https://stackoverflow.com/questions/59520534/lapacke-cheev-only-returns-upper-matrix-of-eigenvectors , it appears that LAPACKE_cheev_work(LAPACK_ROW_MAJOR,'V',...) does not correctly transpose output eigenvectors in version 3.9.0. 

Indeed, it makes use of LAPACKE_che_trans() to do so, but this function is designed to transpose Hermitian matrices and makes use of their upper or lower half depending on uplo. Nevertheless, the orthonormal eigenvectors do not form a Hermitian matrix.

The problem also occurs to other eigenvalues-related routines, such as lapacke_zheev_work.c, lapacke_zheevd_work.c The routines lapacke_zheevr_work.c and lapacke_zheevx_work.c correctly make use of LAPACKE_zge_trans(). Version 3.8.0 is not affected by this issue, as it makes use of LAPACKE_cge_trans() everywhere. Today, the problem only affects LAPACKE 3.9.0.

Here is a sample code displaying this issue:

#include <stdio.h>
#include <complex.h>
#include <lapacke.h>

int main()
{

 lapack_int i,j, n,  lda, info;
 n=4; 

 lda=n;
 float w[n];

 float complex R [n*n];
 for (i=0;i<n;i++){
    for (j=0;j<n;j++){
      R[i*n+j]=0.;
    }
    R[i*n+i]=2.;
    if(i>0){
      R[i*n+(i-1)]=-1;
    }
    if(i<n-1){
      R[i*n+(i+1)]=-1;
    }
 }

 float complex eigenvectors [n*n];
 info= LAPACKE_clacpy(LAPACK_ROW_MAJOR,'U', n,n,R,n,eigenvectors,n); 

 if(info !=0){printf("LAPACKE_clacpy error %d\n",info); }   
 info = LAPACKE_cheev(LAPACK_ROW_MAJOR, 'V', 'U', n, eigenvectors, lda, w);
 if(info !=0){printf("LAPACKE_cheev error %d\n",info); }

 for (i=0;i<n;i++){
    for (j=0;j<n;j++){
        printf("%+6.4f+I*%+6.4f | ",creal(eigenvectors[i*n+j]),cimag(eigenvectors[i*n+j]));
    }
    printf("\n");
 }
 if(creal(eigenvectors[(n-1)*n+0])==0.){
    printf("LAPACKE_cheev : the eigenvectors are wrong due to LAPACKE_che_trans()\n");
    return 1;
 }
 return 0;
}

**Comment by martin-frbg** [comment]
It looks like the author of that change already realized the mistake some months ago, however his PR #330 did not get merged.

**Comment by marvinbernhardt** [comment]
I had the same problem with LAPACKE_ssyev. Took me several hours to find the problem, since everything worked fine three month ago. Should be marked as a bug!

**Comment by weslleyspereira** [comment]
I think #330 solved this and #423. Am I right?
The current output is:
```
-0.3717+I*+0.0000 | +0.6015+I*+0.0000 | +0.6015+I*+0.0000 | -0.3717+I*+0.0000 | 
-0.6015+I*+0.0000 | +0.3717+I*+0.0000 | -0.3717+I*+0.0000 | +0.6015+I*+0.0000 | 
-0.6015+I*+0.0000 | -0.3717+I*+0.0000 | -0.3717+I*+0.0000 | -0.6015+I*+0.0000 | 
-0.3717+I*+0.0000 | -0.6015+I*+0.0000 | +0.6015+I*+0.0000 | +0.3717+I*+0.0000 |
```
and it does not print the error message "LAPACKE_cheev : the eigenvectors are wrong due to LAPACKE_che_trans()\n".

**Comment by marvinbernhardt** [comment]
yes, this seems solved! :+1:
